synapse/replication/tcp/streams/_base.py

```diff
@@ -151,6 +127,26 @@ def update_function(self, from_token, current_token, limit):
         raise NotImplementedError()
 
 
+def db_query_to_update_function(
+    query_function: Callable[[int, int, int], Awaitable[List[tuple]]]
+) -> Callable[[int, int, int], Awaitable[Tuple[List[Tuple[int, tuple]], int, bool]]]:
```

Comment:
feels like we should make some type definitions somewhere...

Reply:
Yeeeeeeeeeeeeeees.

Reply:
Ok, I've added some type aliases, although it didn't really simplify things very much in the end :/

Reply:
well I was really thinking we should have something like:

```
UpdateFunctionResult = Tuple[List[StreamRow], Token, bool]
UpdateFunction = Callable[[Token, Token, int], Awaitable[UpdateFunctionResult]]
```

but shrugface